website/docs/cdktf/concepts/aspects.mdx

```diff
@@ -21,14 +21,22 @@ The TypeScript example below defines an aspect to add tags to resources.
 ```ts
 import { Aspects, IAspect } from "cdktf";
 
+// Not all constructs are taggable, so we need to filter it
+type TaggableConstruct = IConstruct & {
+  tags?: { [key: string]: string };
+};
+function isTaggableConstruct(x: IConstruct): x is TaggableConstruct {
+  return "tags" in x;
+}
+
 export class TagsAddingAspect implements IAspect {
   constructor(private tagsToAdd: Record<string, string>) {}
 
   // This method is called on every Construct within the specified scope (resources, data sources, etc.).
   visit(node: IConstruct) {
-    if (node.hasOwnProperty("tags")) {
+    if (isTaggableConstruct(node)) {
       const currentTags = node.tags || {};
-      node.tags = { ...currentTags, ...this.tagsToAdd };
+      node.tags = Fn.tomap(Fn.merge([currentTags, this.tagsToAdd]));
```

Comment:
This definitely does some strange things under the hood (converting to list token, then to a string token, then casted to any). I'm not 100% sure it ports to other languages very well either.
We should make sure to test in a couple languages and ideally have an example that easily ports (or have separate examples).